src/Features/Lsif/Generator/Generator.cs

```diff
@@ -150,24 +154,25 @@ public async Task GenerateForProjectAsync(Project project, GeneratorOptions opti
             GeneratorOptions options,
             IResultSetTracker topLevelSymbolsResultSetTracker,
             ILsifJsonWriter lsifJsonWriter,
-            IdFactory idFactory)
+            IdFactory idFactory,
+            CancellationToken cancellationToken)
         {
             // Create and keep the semantic model alive for this document.  That way all work/services we kick off that
             // use this document can benefit from that single shared model.
-            var semanticModel = await document.GetRequiredSemanticModelAsync(CancellationToken.None);
+            var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken);
 
-            var (uri, contentBase64Encoded) = await GetUriAndContentAsync(document).ConfigureAwait(false);
+            var (uri, contentBase64Encoded) = await GetUriAndContentAsync(document);
 
             var documentVertex = new Graph.LsifDocument(new Uri(uri, UriKind.RelativeOrAbsolute), GetLanguageKind(semanticModel.Language), contentBase64Encoded, idFactory);
             lsifJsonWriter.Write(documentVertex);
             lsifJsonWriter.Write(new Event(Event.EventKind.Begin, documentVertex.GetId(), idFactory));
 
             // We will walk the file token-by-token, making a range for each one and then attaching information for it
             var rangeVertices = new List<Id<Graph.Range>>();
-            await GenerateDocumentRangesAndLinks(document, documentVertex, options, topLevelSymbolsResultSetTracker, lsifJsonWriter, idFactory, rangeVertices).ConfigureAwait(false);
+            await GenerateDocumentRangesAndLinks(document, documentVertex, options, topLevelSymbolsResultSetTracker, lsifJsonWriter, idFactory, rangeVertices, cancellationToken).ConfigureAwait(false);
```

Comment:
Is there a reason we aren't removing `ConfigureAwait` here? Same question applies two lines below this.

Reply:
just inconsistencies.  hard to get right without analyzer checking everything.

Reply:
removed.